src/issuance.rs

```diff
@@ -365,44 +392,67 @@ impl IssueBundle<Signed> {
 ///     * `AssetBase` for the `IssueAction` has not been previously finalized.
 /// * For each `Note` inside an `IssueAction`:
 ///     * All notes have the same, correct `AssetBase`.
+///
+// # Returns
+///
+/// A Result containing a SupplyInfo struct, which stores supply information in a HashMap.
+/// The HashMap uses AssetBase as the key, and an AssetSupply struct as the value. The
+/// AssetSupply contains a ValueSum (representing the total value of all notes for the asset)
+/// and a bool indicating whether the asset is finalized.
+///
+/// # Errors
+///
+/// * `IssueBundleInvalidSignature`: This error occurs if the signature verification
+///    for the provided `sighash` fails.
+/// * `WrongAssetDescSize`: This error is raised if the asset description size for any
+///    asset in the bundle is incorrect.
+/// * `IssueActionPreviouslyFinalizedAssetBase`:  This error occurs if the asset has already been
+///    finalized (inserted into the `finalized` collection).
+/// * `IssueActionIncorrectAssetBase`: This error occurs if any note has an incorrect note type.
+/// * `ValueSumOverflow`: This error occurs if an overflow happens during the calculation of
+///     the value sum for the notes in the asset.
+/// * `IssueBundleIkMismatchAssetBase`: This error is raised if the `AssetBase` derived from
+///    the `ik` (Issuance Validating Key) and the `asset_desc` (Asset Description) does not match
+///    the expected `AssetBase`.
 pub fn verify_issue_bundle(
     bundle: &IssueBundle<Signed>,
     sighash: [u8; 32],
     finalized: &mut HashSet<AssetBase>, // The finalization set.
-) -> Result<(), Error> {
-    if let Err(e) = bundle.ik.verify(&sighash, &bundle.authorization.signature) {
-        return Err(IssueBundleInvalidSignature(e));
-    };
+) -> Result<SupplyInfo, Error> {
```

Comment:
following our discussion I realized that the signature not entirely matches our spec. We need to make the following changes: 
- Remove `mut` from the `finalized` HashSet. - Should not changed during the execution.
- Add the ability to get the new finalized set from `SupplyInfo`.

The rational here is to have the ability to verify multiple issue bundles for the same block. Only once we finished verifying all the TXs in the block, we should update the finalization set for the next block, not for the next TX.

Reply:
Fixed it in a separate commit.